src/lib/ens.ts

```diff
@@ -0,0 +1,40 @@
+// @ts-ignore TODO: add type declarations for ensjs module
+import ENS, { getEnsAddress } from '@ensdomains/ensjs'
+import memoize from 'lodash/memoize'
+import { getWeb3Provider } from 'lib/web3-provider'
+
+let maybeEns: ENS | null
+
+const web3Provider = getWeb3Provider()
+
+const getEnsInstance = (): ENS => {
+  if (!maybeEns) {
+    // getEnsAddress takes a magic number as string, networkId. 1 stands for mainnet
+    maybeEns = new ENS({ provider: web3Provider, ensAddress: getEnsAddress('1') })
+    return maybeEns!
+  } else {
+    return maybeEns!
+  }
+}
```

Comment:
This is like half bikeshedding, but note that it means you can do `await getWeb3Provider()` safely -- I want to make that async later so that ability would be nice

```suggestion
let makeEns: () => void
// getEnsAddress takes a magic number as string, networkId. 1 stands for mainnet
const ens = new Promise<void>(resolve => makeEns = resolve).then(async () => {
  return new ENS({ provider: await getWeb3Provider(), ensAddress: getEnsAddress('1') })
})
```